pkgs/development/python-modules/google-crc32c/default.nix

```diff
@@ -0,0 +1,26 @@
+{ lib, pkgs, buildPythonPackage, fetchPypi, cffi }:
+
+buildPythonPackage rec {
+  pname = "google-crc32c";
+  version = "1.0.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "lDm5YLbs2EdVdnXRMPw2Jtdiv1NdpZXCCmlJpwX7Pq4=";
+  };
+
+  buildInputs = [ pkgs.crc32c  ];
+  propagatedBuildInputs = [ cffi ];
+
+  preConfigure = ''
+    export LDFLAGS="-L${pkgs.crc32c}/lib"
+    export CFLAGS="-I${pkgs.crc32c}/include"
+  '';
+
```

Comment:
Sorry, we're probably overwhelming you here but if you do

```
  src = fetchFromGitHub {
    owner = "googleapis";
    repo = "python-crc32c";
    rev = "v${version}";
    sha256 = "0n3ggsxmk1fhq0kz6p5rcj4gypfb05i26fcn7lsawakgl7fzxqyl";
  };

  ...

  checkInputs = [ pytestCheckHook ];
  pythonImportsCheck = [ "google_crc32c" ];
```

we should get the tests working. We should _at least_ have the `pythonImportsCheck` otherwise we have zero warning that this package is broken.

Reply:
no probem always love to add more tests

Reply:
hum it broke the tests, apparently the c library is no more found

Reply:
it looks like it is the LDFLAGS/CFLAGS changes you proposed

Reply:
ok it is the pytestCheckHook that fails, it cannot load the module when this hook is enabled

Reply:
That's the hook that actually runs the tests. It's weird because it works fine on macos, fails on linux.

It's the same for me however the `LDFLAGS`/`CFLAGS` are set.

Reply:
does it make sense to only use the hook on darwin ?

Reply:
Yeah if you look, the `pythonImportsCheck` also gives the warning:

```
Check whether the following modules can be imported: google_crc32c
/nix/store/avd5d8ypvjfmrsp9q7b2v2cz6bldlbzp-python3.7-google-crc32c-1.0.0/lib/python3.7/site-packages/google_crc32c/__init__.py:29: RuntimeWarning: As the c extension couldn't be imported, `google-crc32c` is using a pure python implementation that is significantly slower. If possible, please configure a c build environment and compile the extension
```

so I suspect this was at no point successfully compiling a _working_ binary, and was always just falling back to the pure python implementation.

Reply:
It's not the hook that's causing the problem, it's that it's exposing that the build is broken.

Reply:
Looks like the generated so is missing a reference to this function

`ImportError: /nix/store/r9sa0sqljf62qjhr9k4fifwq66b0kbjm-python3.8-google-crc32c-1.0.0/lib/python3.8/site-packages/google_crc32c/_crc32c_cffi.abi3.so: undefined symbol: __cxa_guard_release`

But I don't understand which library it is related to ...

Reply:
Of course, if we just want to get this over the wall and move on with the release, we can leave a big `TODO` here: "compile against `pkgs.crc32c` for expected performance gain" and turn off the tests leaving just the `pythonImportsCheck`, because it does _work_ without the compiled extension.

If we're going to do that though there's no point in providing it with `pkgs.crc32c` or setting any special `LDFLAGS` etc., it would just be confusing.

Reply:
> But I don't understand which library it is related to ...

It's a c++11 thing, it's just a little confusing why it's not being linked in automagically as we should be providing it with all it needs... :thinking:

Reply:
Oh _that's_ very interesting - our `pkgs.crc32c` package on linux doesn't produce any dynamic `.so` libs, only static `libcrc32c.a`.

Looks like it's the same on macos, but for some reason `clang` will link in c++ requirements whereas `gcc` won't. (?)

Reply:
we can enable the shared lib

```
if(BUILD_SHARED_LIBS)
  set_property(TARGET crc32c_sse42 PROPERTY POSITION_INDEPENDENT_CODE TRUE)
endif(BUILD_SHARED_LIBS)
```

Reply:
Oh .. I tried to force c++11 and it told me it is C, not C++, that must be the problem

Reply:
Yup we're finding this in parallel - it works if we set `cmakeFlags = [ "-DBUILD_SHARED_LIBS=1" ];`. The annoying thing is it won't build both the static and shared libs at the same time, and I'm not sure we want to take away the static lib from packages that might be happily using it. Just checking how many packages actually use it...

Reply:
I think what we do is add a `shared` argument to `pkgs/development/libraries/crc32c/default.nix`, defaulting to `false`, which just does a `cmakeFlags = optional shared "-DBUILD_SHARED_LIBS=1";`. Then we just pass `crc32c.override { shared = true; }` to this package in `python-packages.nix`.

Thoughts @jonringer ?

Reply:
usually you opt into static.

I would add a `staticOnly ? false` input. 

Then in `pkgs/top-level/static.nix` you would add:
```
  crc = super.crc.override {
    staticOnly = true;
  };
```

that way, when you do `pkgsStatic.crc` you get the `.a`, and normally you get the `.so`

Reply:
applied these fixes and built locally successfully :tada: :

 * pkgs.crc32c
 * pkgsStatic.crc32c
 * pkgs.python2Packages.google_resumable_media
 * pkgs.python37Packages.google_resumable_media
 * pkgs.python38Packages.google_resumable_media